### PR TITLE
Permit .jpeg images to be uploaded straight into description fields. #163

### LINKED ISSUE #163: Permit .jpeg images to be uploaded straight into description fields.

### PR DESCRIPTION
**Pull request by DoxasticFox**
Is this what you had in mind, @rosiejbrown?

### User-facing:

Screenshot from 2023-08-24 10-35-23

### Admin-facing:

Screenshot from 2023-08-24 10-37-31

---

Closes #163

### CODE REVIEW

**Comment by rosiejbrown** [comment]
@DoxasticFox yes, it looks great. Two questions:
1. Does this format mean the picture always sits at the bottom of the text in either the description or the question part? I think that's fine if the case, just curious.
2. Will we also have this option in just the description parts of CTRL? So the option to add an image will be available for the starting paragraphs and when you select a description only text?

**Comment by DoxasticFox** [comment]
>Does this format mean the picture always sits at the bottom of the text in either the description or the question part? I think that's fine if the case, just curious.

Yep. It can be made customisable, but only with lots of work. Not sure how much work without looking into it, but I assume it'd be more than a week.

>Will we also have this option in just the description parts of CTRL? So the option to add an image will be available for the starting paragraphs and when you select a description only text?

For the benefit of anyone reading, as discussed on Slack, this PR doesn't currently let the admin add images to the description in each step. But that's easy to add, so I'll add it now :slightly_smiling_face:

File: app/assets/stylesheets/pages/_steps.sass [hunk 1]
```diff
@@ -93,3 +92,19 @@ $component: 'steps'
...
+        max-width: 100%
+        margin-bottom: 20px
+
+    +element('description-image-container')
+        display: flex
+        justify-content: center
+
+    +element('description-image')
+        max-width: 100%
+        margin-top: 20px
```
Comment (DoxasticFox) [inline app/assets/stylesheets/pages/_steps.sass:110]: Used by `app/javascript/consent-form.vue`

File: app/models/consent_step.rb [hunk 2]
```diff
@@ -10,6 +10,10 @@ class ConsentStep < ApplicationRecord
   # buggy with has_one associations
   has_many :modal_fallbacks, dependent: :destroy
 
+  has_one_attached :description_image
+
+  attr_accessor :remove_description_image
```
Comment (DoxasticFox) [inline app/models/consent_step.rb:15]: All the `attr_accessor` method calls are required to make images deleteable via Active Admin

File: db/migrate/20230823055423_create_active_storage_tables.active_storage.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  private
+    def primary_and_foreign_key_types
+      config = Rails.configuration.generators
+      setting = config.options[config.orm][:primary_key_type]
+      primary_key_type = setting || :primary_key
+      foreign_key_type = setting || :bigint
+      [primary_key_type, foreign_key_type]
+    end
+end
```
Comment (DoxasticFox) [inline db/migrate/20230823055423_create_active_storage_tables.active_storage.rb:57]: This migration was generated automatically by running `rails active_storage:install`

**Review by ammazzaw — APPROVED** [review]

**Pull request merged**